pkg/proxyserver/getter/logProxyGetter.go

```diff
@@ -0,0 +1,115 @@
+package getter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/stolostron/multicloud-operators-foundation/pkg/helpers"
+	"io"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"net/http"
+	clusterv1listers "open-cluster-management.io/api/client/cluster/listers/cluster/v1"
+)
+
+type LogProxyGetter struct {
+	ClusterLister      clusterv1listers.ManagedClusterLister
+	SecretLister       corev1listers.SecretLister
+	ProxyServiceHost   string
+	ProxyServiceCAFile string
+}
+
+func NewLogProxyGetter(clusterLister clusterv1listers.ManagedClusterLister,
+	secretLister corev1listers.SecretLister,
+	host, caFile string) *LogProxyGetter {
+	return &LogProxyGetter{
+		ClusterLister:      clusterLister,
+		SecretLister:       secretLister,
+		ProxyServiceHost:   host,
+		ProxyServiceCAFile: caFile,
+	}
+}
+
+func (c *LogProxyGetter) ProxyServiceAvailable(clusterName string) (bool, error) {
+	cluster, err := c.ClusterLister.Get(clusterName)
+	if err != nil {
+		return false, fmt.Errorf("failed to get cluster %s. %v", clusterName, err)
+	}
+	labels := cluster.GetLabels()
+	if len(labels) != 0 &&
+		labels[helpers.MsaAddOnFeatureLabel] == helpers.AddOnStatusAvailable &&
+		labels[helpers.ClusterProxyAddOnFeatureLabel] == helpers.AddOnStatusAvailable {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *LogProxyGetter) NewHandler(clusterName, podNamespace, podName, containerName string) (*Handler, error) {
+	logTokenSecret, err := c.SecretLister.Secrets(clusterName).Get(helpers.LogManagedServiceAccountName)
```

Comment:
I am not sure we need a secret lister. The handler is not a very frequent call. Directly get should be good enough IMO

Reply:
changed to use SecretLister here